### PR TITLE
(MODULES-3203) Add Darwin support

### PR DESCRIPTION
**Pull request by keeleysam**
This adds Darwin support, it is very similar to FreeBSD.

### CODE REVIEW

**Comment by keeleysam** [comment]
I need to fix a couple of the tests

File: metadata.json [hunk 1]
```diff
@@ -98,6 +98,18 @@
         "2014.09",
         "2015.03"
       ]
+    },
+    {
+      "operatingsystem": "Darwin",
+      "operatingsystemmajrelease": [
+        "9",
```
Comment (igalic) [inline metadata.json]: officially puppetlabs supported?

Comment (keeleysam) [inline metadata.json]: Updated to be Darwin 13+ which is the versions supported by PE.

**Comment by jonnytpuppet** [comment]
Hi There

Thanks for your contribution!

In order for us to merge this, it would be appreciated if you could have a look at why the tests are failing. It's quite possible that it's related to something outside of this PR, though

Thanks

**Comment by tphoney** [comment]
Hi @keeleysam, thanks for the work you have put into this pull request. However due to inactivity and failing tests / needing rebased, this pull request is being closed. If you want to re-submit this PR with the necessary changes, we will gladly look at the new pull request.

**Pull request closed without merging**